$OpenBSD: patch-libavcodec_aacenc_pred_c,v 1.1 2015/09/12 09:44:54 ajacoutot Exp $

aacenc_pred: disable PNS SFBs and disable right predictors with IS

--- libavcodec/aacenc_pred.c.orig	Sat Sep 12 02:08:04 2015
+++ libavcodec/aacenc_pred.c	Sat Sep 12 02:08:25 2015
@@ -262,7 +262,9 @@ void ff_aac_search_for_pred(AACEncContext *s, SingleCh
         const int num_coeffs = sce->ics.swb_offset[sfb + 1] - start_coef;
         const FFPsyBand *band = &s->psy.ch[s->cur_channel].psy_bands[sfb];
 
-        if (start_coef + num_coeffs > MAX_PREDICTORS)
+        if (start_coef + num_coeffs > MAX_PREDICTORS ||
+            (s->cur_channel && sce->band_type[sfb] >= INTENSITY_BT2) ||
+            sce->band_type[sfb] == NOISE_BT)
             continue;
 
         /* Normal coefficients */
